float32 total_voltage
float32 current
float32 soc_percent
float32[] cell_voltages
int8 charging_mosfet
int8 discharging_mosfet
float32 capacity_ah
string[] error_strings